### PR TITLE
[4.0] Rewrite stats in vanilla JS

### PR DESCRIPTION
**Pull request by C-Lodder**
### Summary of Changes

- Rewrite the stats script in vanilla JS
- The the proper CE event to close alerts
- Throw an alerts when unable to connect to `developer.joomla.org`

### Testing Instructions

- Fresh installation or reset the Stats plugin to that you get a nice big blue alert.
- Click the link inside the text to display which stats will be submitted.
- Click on any of the 3 buttons

### CODE REVIEW

File: media/plg_system_stats/js/stats.js [hunk 1]
```diff
@@ -1,93 +1,113 @@
...
+		Joomla.request({
+			url: 'index.php?option=' + data.option + '&group=' + data.group + '&plugin=' + data.plugin +  '&format=' + data.format,
+			method: 'GET',
+			perform: true,
+			headers: {'Content-Type': 'application/json'},
+			onSuccess: function(response, xhr) {
+				try {
+					response = JSON.parse(response);
+				} catch(e) {
+					console.log(e);
```
Comment (dgrammatiko) [inline media/plg_system_stats/js/stats.js]: Can you replace the console log with:
```js
throw new Error(e);
```

**Comment by C-Lodder** [comment]
@dgt41 done

**Comment by brianteeman** [comment]
You removed the header stating that the js is for and its version - was that deliberate?

**Comment by C-Lodder** [comment]
@brianteeman deliberate. I dont see that comment block in any other JS file so keep them consistent

**Review by dgrammatiko — APPROVED** [review]

**Comment by brianteeman** [comment]
it is present in 36/53 js files in j4 - my view is that it should be present in them all

**Comment by dgrammatiko** [comment]
I agree with @brianteeman 
The hard part is to get the since in the old scripts that don't have it

**Comment by mbabker** [comment]
> The hard part is to get the since in the old scripts that don't have it

Don't worry about them.  Just put a description in the doc block and be done with it.  There are already plenty of PHP endpoints with bad tags, unless you really want to waste a Saturday chasing Git/SVN history leave it be.

**Comment by brianteeman** [comment]
I was only suggesting keeping the headers not searching for old ones ;)

**Pull request merged**